Add task to ensure that download+install dirs exists.

ingomueller-net opened the pull request:
This may be interesting if Spark is to be installed in a non-standard
location, for example alongside other versions in /opt.

ingomueller-net committed:
Add task to ensure that download+install dirs exists.

This may be interesting if Spark is to be installed in a non-standard
location, for example alongside other versions in /opt.

hectcastro commented on tasks/main.yml:
```diff
@@ -23,6 +23,14 @@
     - "{{ spark_log_dir }}"
     - "{{ spark_run_dir }}"
 
+- name: Ensure Spark download and install directories exist
```
Does it make sense to append `spark_src_dir` and `spark_usr_parent_dir` to the task above this, or is the idea to keep these two directories so that `owner` and `group` aren't owned by `spark`?

ingomueller-net replied:
Yes, I think it is slightly cleaner if the two are different tasks. My next commit also gives write access to the ```spark``` group (such that other users can start and stop the cluster), which I guess we wouldn't want to be the case for the code.

ingomueller-net committed:
Extend write permission of Spark directories to Spark group.

This may be required if several users should be allowed to start/stop
the cluster.

ingomueller-net committed:
Create Spark work and temp directories, which are needed for slaves.

ingomueller-net commented:
Repeating my comment from [the outdated conversation](https://github.com/azavea/ansible-spark/pull/21/files/7a752a2c45e684f4820fd358dbcdf44a9df54afe#diff-2444ad0870f91f17ca6c2a5e96b26823):

> Yes, I think it is slightly cleaner if the two are different tasks. My next commit also gives write access to the spark group (such that other users can start and stop the cluster), which I guess we wouldn't want to be the case for the code.

hectcastro commented on tasks/main.yml:
```diff
@@ -13,15 +13,24 @@
   with_items:
     - "{{ spark_log_dir }}"
     - "{{ spark_run_dir }}"
+    - "{{ spark_work_dir }}"
```
This addition appears to be causing a problem when provisioning from scratch. Because a directory is being created inside of `/usr/lib/spark`, the first item in `Setup Spark distribution symlinks` is failing.

ingomueller-net commented:
Ah, right, good catch! I see two ways out: (1) either define set `spark_work_dir: "{{ spark_usr_parent_dir }}/spark-{{ spark_version }}"` in `defaults.yml` (with the disadvantage that it can break if someone sets it to the current value), or (2) use another task to create the work directory after the symlinks have been created (with the disadvantage of duplicating code). Preferences from your side?

hectcastro commented:
@ingomueller-net, thanks for outlining the options. I think 2 is probably the safest.

ingomueller-net commented:
OK, MR updated. Sorry for taking a while. I also added a `tmp` directory, which I have found useful.

hectcastro commented:
Looks good. Thank you, @ingomueller-net!
